### PR TITLE
fix(dashboard): Inoperable dashboard filter slider when range is <= 1

### LINKED ISSUE #24010: Inoperable dashboard filter slider when range is <= 1
`antd`'s slider, as used in dashboard filters, has a step size of 1 (the default).  So when the entire range of numbers is within 1, the slider is inoperable.

#### How to reproduce the bug

1. Create a chart using a dataset that has a column whose values are within a range of `1`.   E.g. a column all of whose values are between 0 and 1.
1a. If you don't have such a dataset, edit a dataset using "calculated columns" to create one.
2. Create a dashboard that uses that chart.
3. Add a numerical range filter to that dashboard, connected to the column in step (1).
4. Try to use the slider

### Expected results

The slider should slide e.g. between 0 and 1

### Actual results

Since the slider's step size is `1` the slider doesn't slide

### Environment

- superset version: `2.1.0`

### Checklist

Make sure to follow these steps before submitting your issue - thank you!

- [ ] I have checked the superset logs for python stacktraces and included it here as text if there are any.
- [x] I have reproduced the issue with at least the latest released version of superset.
- [x] I have checked the issue tracker for the same issue and I haven't found one similar.

### Additional context

This seems to be related:
https://github.com/apache/superset/pull/18009

### PR DESCRIPTION
**Pull request by jfrancos-mai**
### SUMMARY
`antd`'s slider, as used in dashboard filters, has a step size of 1 (`antd`'s default). So when the entire range of numbers is within 1, the slider is inoperable.

A few possible solutions -- I chose option (3):

1. Set the step size to something static that fixes this particular issue, like .01.  This makes things better for this case, but worse for cases where a step size of `1` is sufficient.
2. Let the user choose a step size in the filter setup dialog.  Nice because it makes things explicit, but adds extra UI stuff and users have to make extra decisions.
3. Come up with a heuristic for choosing a step size.  Nice because it works by default, and without any extra work from the user.  If we want the user to be able to choose their own step size, that could be a feature rather than a bug fix (and seems to be in-progress [over here](https://github.com/apache/superset/pull/18009
))

This solution preserves the default behavior where the rightmost digit moves up and down by 1, while accounting for the cases where that 1 ought to be after the decimal point. The calculated step-size is the largest such number (but not larger than `1`) that accommodates the minimum number of steps we'd want to have (`MIN_NUM_STEPS`)

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

https://github.com/apache/superset/assets/99757211/1f513f77-2c30-4c64-baff-088d3fcc236b

https://github.com/apache/superset/assets/99757211/72188e8b-0e76-4a23-ab5b-d9709072532a

#### How to reproduce the bug

1. Create a chart using a dataset that has a column whose values are within a range of `1`.   E.g. a column all of whose values are between 0 and 1.
1a. If you don't have such a dataset, edit a dataset using "calculated columns" to create one.
2. Create a dashboard that uses that chart.
3. Add a numerical range filter to that dashboard, connected to the column in step (1).
5. Try to use the slider

### ADDITIONAL INFORMATION

Fixes #24010
- [ ] Has associated issue:
- [ ] Required feature flags:
- [x] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by rusackas** [comment]
Pinged a couple of reviewers, but I'm a little nervous that this might cause as many problems as it solves. If your data has a range of integers from 1-10, for example, then this would cause 20 steps in that range, and may cause queries to fail. 

If this is intended specifically for values between 0 and 1, maybe an agreeable compromise is:
`const step = max <= 1 ? stepHeuristic(min, max) : 1;`

**Comment by jfrancos-mai** [comment]
> Pinged a couple of reviewers, but I'm a little nervous that this might cause as many problems as it solves. If your data has a range of integers from 1-10, for example, then this would cause 20 steps in that range, and may cause queries to fail.
> 
> If this is intended specifically for values between 0 and 1, maybe an agreeable compromise is: `const step = max <= 1 ? stepHeuristic(min, max) : 1;`

Hi @rusackas, happy to make that adjustment but a couple questions about it:
> `const step = max <= 1 ? stepHeuristic(min, max) : 1;`

Did you mean:
`const step = max - min <= 1 ? stepHeuristic(min, max) : 1;`?

Also, I do like
```
const step = max - min <= 1 ? stepHeuristic(min, max) : 1;
```
but wondering if you wouldn't prefer
```
const step = max - min <= 1 ? stepHeuristic(min, max) : undefined;
```
(where `undefined` should result in antd's default value of `1`).   I agree that using `1` instead of `undefined` is better, as explicit is better than implicit, but maybe it's better to make that change as part of a separate issue?

**Comment by michael-s-molina** [comment]
Thank you for the PR @jfrancos-mai! I think the explicit version is preferable:

`const step = max - min <= 1 ? stepHeuristic(min, max) : 1;`

File: superset-frontend/src/filters/components/Range/RangeFilterPlugin.tsx [hunk 1]
```diff
@@ -300,6 +300,16 @@ export default function RangeFilterPlugin(props: PluginFilterRangeProps) {
...
 
+  const MIN_NUM_STEPS = 20;
+  const stepHeuristic = (min: number, max: number) => {
+    const maxStepSize = (max - min) / MIN_NUM_STEPS;
+    // normalizedStepSize: .06 -> .01, .003 -> .001
+    const normalizedStepSize = `1E${Math.floor(Math.log10(maxStepSize))}`;
+    return Math.min(1, parseFloat(normalizedStepSize));
+  };
+
+  const step = stepHeuristic(min, max);
```
Comment (michael-s-molina) [inline superset-frontend/src/filters/components/Range/RangeFilterPlugin.tsx:311]:
```suggestion
  const step = max - min <= 1 ? stepHeuristic(min, max) : 1;
```

Comment (michael-s-molina) [inline superset-frontend/src/filters/components/Range/RangeFilterPlugin.tsx:311]:
```suggestion
  const step = max - min <= 1 ? stepHeuristic(min, max) : 1;
```

**Comment by rusackas** [comment]
@jfrancos-mai just wondering if you still want to follow through on this PR. I hope so, but if not, maybe we should close it, or at least convert it to a draft if you want to come back to it at some point in the future.

**Comment by michael-s-molina** [comment]
Closing in favor of https://github.com/apache/superset/pull/27271.

**Pull request closed without merging**